.github/workflows/test.migration.yaml

```diff
@@ -0,0 +1,127 @@
+name: "Test - Migration"
+on:
+  pull_request:
+    branches: [rc, main]
+  push:
+    branches: [rc, main]
+  workflow_dispatch:
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      changed: ${{ steps.filter.outputs.changed }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Diff Changes
+        uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            changed:
+              - "client/**"
+              - "freighter/py/**"
+              - "synnax/**"
+              - "cesium/**"
+              - "aspen/**"
+              - "x/**"
+              - "freighter/**"
+              - ".github/workflows/test.client.yaml"
+
+  test-on-both-servers:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      # Test with Main Server
```

Comment:
`Main` and `Current` are bad names. Something like "Test against current production version (main branch)", and "Test against release candidate (rc branch)" would be clearer.